### PR TITLE
Test to run gdb testsuite - v3

### PR DESCRIPTION
**Pull request by Pavithra1602**
link to v2 - https://github.com/avocado-framework/avocado-misc-tests/pull/108

Change log:
1. Removed check for packages.
2. Modified coding style in assigning packages list.

### CODE REVIEW

**Comment by apahim** [comment]
Checking.

File: generic/gdb.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        self.srcdir = os.path.join(
+            self.srcdir, os.path.basename(tarball.split('.tar')[0]))
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
+    def test(self):
+        process.run("make check-gdb", ignore_status=True, sudo=True)
+        logfile = os.path.join(self.logdir, "stdout")
+        fp = open(logfile, "r")
```
Comment (apahim) [inline generic/gdb.py:57]: This is keeping the file open and then avocado cannot remove the srcdir after the test:
It is a best practice to use `with` to deal with file objects, so the file is always closed after the commands inside the with....

```
with open(file, 'r') as f:
    for line in f.readlines():
...
```

**Comment by Pavithra1602** [comment]
Raised new PR #114.
Closing this Pull request.

Thanks,
Pavithra

File: generic/gdb.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import distro
+from avocado.utils import process
```
Comment (PraveenPenguin) [inline generic/gdb.py:24]: can not be do like this way 
from avocado.utils import distro, archive, process, build

**Pull request closed without merging**